src/common/LogClient.cc

```diff
@@ -163,9 +163,15 @@ void LogChannel::do_log(clog_type prio, const std::string& s)
 {
   std::lock_guard l(channel_lock);
   if (CLOG_ERROR == prio) {
-    ldout(cct,-1) << "log " << prio << " : " << s << dendl;
+    ldout(cct, -1) << "log " << prio << " : " << s << dendl;
+  } else if (CLOG_WARN == prio) {
+    ldout(cct, 0) << "log " << prio << " : " << s << dendl;
+  } else if (CLOG_SEC == prio) {
+    ldout(cct, 4) << "log " << prio << " : " << s << dendl;
+  } else if (CLOG_INFO == prio) {
+    ldout(cct, 9) << "log " << prio << " : " << s << dendl;
   } else {
-    ldout(cct,0) << "log " << prio << " : " << s << dendl;
+    ldout(cct, 20) << "log " << prio << " : " << s << dendl;
```

Comment:
> In code we found that except CLOG_ERROR all log messages get ceph priority level 0 which maps to syslog/journald priority level 4.

I doubt `ldout` log level influences the syslog/journald log level in any way.   AFAIU these `ldout` statements are there to reflect the messages that are sent to the cluster log in the regular log as well.  `CLOG_ERROR` is mapped to `-1` and everything else is mapped to `0` to ensure that everything that is sent to the cluster log also shows up in the regular (daemon) log independent of what `debug mon` or similar options are set to.  With 0 changed to 4, 9 or 20, that would break because e.g. `debug mon = 9` setting would be required to see `CLOG_INFO` messages in the regular log.

BTW where did 9 come from?  Doesn't syslog priority/severity go only up to 7, with `Informational` being 6 and `Debug` being 7?

The syslog/journald log level for cluster log messages is supposed to be controlled by `clog_to_syslog_level` config option.  It defaults to `info`, but if you have it set to `warn`, I think it's expected that everything is mapped to 4.  If the objective is to map `CLOG_` priorities to syslog priorities, you would probably need to change
```
  // log to syslog?
  if (do_log_to_syslog()) {
    ldout(cct,0) << __func__ << " log to syslog"  << dendl;
    e.log_to_syslog(get_log_prio(), get_syslog_facility());
  }
```
below, not these `ldout` statements.

Adding @batrick to keep me honest.

Reply:
@idryomov 
Here is the thing, this log statements will go into syslog or journald if you enable `log_to_journald` or `log_to_syslog`, 
`dout_impl` calls `_dout_cct->_log->submit_entry(std::move(_dout_e)); ` internally, if you check the function `void Log::_flush(EntryVector& t, bool crash)` you will get this,
```
if (do_journald && m_journald) {
      m_journald->log_entry(e);
    }
```
Now question is what should be the priority level, ldout(cct, x) here x does not represent syslog priority level, it actually ends up in `CEPH_PRIO` field of journald, in `Journald.cc` you will get the mapping between `CEPH_PRIO` and `PRIORITY_FIELD`
```
char map_prio(short ceph_prio)
{
  if (ceph_prio < 0)
    return LOG_ERR;
  if (ceph_prio == 0)
    return LOG_WARNING;
  if (ceph_prio < 5)
    return LOG_NOTICE;
  if (ceph_prio < 10)
    return LOG_INFO;
  return LOG_DEBUG;
}
```

Reply:
https://www.reddit.com/r/ceph/comments/1g4hjk9/taming_ceph_logging_journal_priorities_out_of/?rdt=40703
Here is some complaints

Reply:
It's possible I'm also confused but I think this `src/common/LogClient.h` code has little at all to do with the `Log::_flush` code paths. This code is concerned with logging to the Ceph cluster log. Cluster log messages are not common so I doubt this has much to do with the problem discussed in the reddit thread. Still, that doesn't mean we shouldn't reevaluate this code. I am also concerned that this may break some testing which (probably wrongly) relies on detecting cluster log messages from the daemon's debug log.

Overall, I'm uncertain mapping cluster log message priorities to debug levels is worthwhile. I would suggest selecting one priority and sticking with it. Admins should be monitoring the cluster log for those messages and not the debug logs of the daemon which submitted it. Any change requires testing across components to look for breakage.

Reply:
Got it, but do you think 0 is level we should go for as it maps to WARNING in journald?

Reply:
I'm fine with using `1` or any other level (I haven't checked what it is) that doesn't produce warnings in the journal/syslog.

Reply:
If I put ceph prio level to 5 it will map to LOG_INFO in journald which seems suitable for me.